### PR TITLE
Add models for all tables

### PR DESCRIPTION
**Pull request by jaferIdrees**
### On this branch completed the below tasks:

- [ ] Added models for User, Post, Comment, and Like tables
- [ ] Users: A method that returns the 3 most recent posts for a given user.
- [ ] Posts: A method that updates the posts counter for a user.
- [ ] Posts: A method which returns the 5 most recent comments for a given post.
- [ ] Comments: A method that updates the comments counter for a post.
- [ ] Likes: A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', counter_cache: :posts_counter
+  has_many :comments
+
```
Comment (thecodechaser) [inline app/models/post.rb]: - [x] Please set up associations between Post and Like. `(Post has many likes)`. You can consider the below code.

```ruby
has_many :likes

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @jaferIdrees👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 1 above)
Comment (jaferIdrees) [reply app/models/post.rb]: Done

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+class Like < ApplicationRecord
```
Comment (rloterh) [inline app/models/like.rb:1]: - [ ] OPTIONAL
`Like` model should have method that updates the likes counter for a post. This should be similar to the example below:

```ruby
# app/models/like.rb
...

def update_likes_counter
  post.increment!(:likes_counter)
end
```

File: app/models/post.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class Post < ApplicationRecord
```
Comment (rloterh) [inline app/models/post.rb:1]: - [ ] OPTIONAL
`Post` model should have method that updates the posts counter for a post. This should be similar to the example below:

```ruby
# app/models/like.rb
...

def update_post_counter
  author.increment!(:posts_counter)
end

```

File: app/models/comment.rb [hunk 4]
```diff
@@ -0,0 +1,4 @@
+class Comment < ApplicationRecord
```
Comment (rloterh) [inline app/models/comment.rb:1]: - [ ] OPTIONAL
Comment model should have a method that updates the comments counter for a post, similar to the example below:

```ruby
# app/models/comment.rb
...

def update_comments_counter
  post.increment!(:comments_counter)
end
```

File: app/models/like.rb (hunk 2 above)
Comment (rloterh) [inline app/models/like.rb:1]: - [ ] OPTIONAL
It is a nice idea to use one of the Rails Active Callbacks `after_save` to call the update_likes_counter method every time a new like has been saved to database, then make the `update_likes_counter` method private, similar to:

```ruby
 # app/models/likes.rb
  after_save :update_like_counter

  private

  def update_like_counter
    author.increment!(:likes_counter)
  end
```

File: app/models/post.rb (hunk 3 above)
Comment (rloterh) [inline app/models/post.rb:1]: - [ ] OPTIONAL
It is a nice idea to use one of the Rails Active Callbacks `after_save` to call the `update_posts_counter method every time a new like has been saved to database, then make the `update_likes_counter` method private, similar to:

```ruby
 # app/models/post.rb
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

File: app/models/comment.rb (hunk 4 above)
Comment (rloterh) [inline app/models/comment.rb:1]: - [ ] OPTIONAL
It is a nice idea to use one of the Rails Active Callbacks `after_save` to call the `update_comments_counter method every time a new like has been saved to database, then make the `update_likes_counter` method private, similar to:

```ruby
 # app/models/comment.rb
  after_save :update_comment_counter

  private

  def update_comment_counter
    author.increment!(:comments_counter)
  end
```

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Jafar,  

Your project is complete! There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 However, kindly consider my comments marked _OPTIONAL_.

## Highlights 👍
- Descriptive Pull Request
- No linter errors

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

_Kindly consider my comments marked **[OPTIONAL]** under this review_

**You can also consider:**
- I think you should change the header title _Authors_ to _Author_ in your ReadMe, as this project is worked on by a sole contributor. 
- Demo link in your ReadMe is broken, kindly endeavor to fix it.
- MIT license link in your ReadMe is broken, kindly endeavor to fix it.
Implementing the above change(s) will make your ReadMe/ repository stand out even better 😉

Keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/like.rb (hunk 2 above)
Comment (jaferIdrees) [reply app/models/like.rb:1]: Hi @rloterh,
Many thanks for your comment.
But I already applied this requirement using (counter_cache: :likes_counter) which should be more efficient.
Do you think I need to change that?

Comment (rloterh) [reply app/models/like.rb:1]: Hey @jaferIdrees 
That's good implementation from you! Cheers! 😃

**Pull request closed without merging**